src/Prismic/SearchForm.php

```diff
@@ -63,157 +82,162 @@ public function getData()
+            $data[$key] = is_array($data[$key]) ? $data[$key] : [$data[$key]];
+            $data[$key][] = $value;
+        } else {
+            $data[$key] = $value;
+        }
+
+        return new self($this->client, $this->cache, $this->form, $data);
     }
 
     /**
-     * Set the repository's ref.
+     * Set the repository ref to query at
      *
-     * @param  string|\Prismic\Ref $ref the ref we wish to query on, or its ID.
-     *
-     * @return Prismic::SearchForm a clone of the SearchForm object with the new ref parameter added
+     * @param  string|Ref $ref the ref we wish to query on, or its ID.
+     * @return self
      */
-    public function ref($ref)
+    public function ref($ref) : self
     {
-        if ($ref instanceof \Prismic\Ref) {
-            $ref = $ref->getRef();
+        if ($ref instanceof Ref) {
+            $ref = (string) $ref;
         }
         return $this->set("ref", $ref);
     }
 
     /**
      * Set the after parameter: the id of the document to start the results from (excluding that document).
-     *
      * @param string $documentId
-     *
-     * @return Prismic::SearchForm a clone of the SearchForm object with the new after parameter added
+     * @return self
      */
-    public function after($documentId)
+    public function after(string $documentId) : self
     {
         return $this->set("after", $documentId);
     }
 
     /**
-     * Set the fetch parameter: restrict the fields to retrieve for a document. You can pass in parameter
-     * an array of strings, or several strings.
+     * Set the fetch parameter: restrict the fields to retrieve for a document
+     *
+     * Pass multiple string arguments or an array of strings to unpack with the splat operator
      *
-     * @return Prismic::SearchForm a clone of the SearchForm object with the new fetch parameter added
+     * @param string[] $fields
+     * @return self
      */
-    public function fetch()
+    public function fetch(string ...$fields) : self
     {
-        $numargs = func_num_args();
-        if ($numargs == 1 && is_array(func_get_arg(0))) {
-            $fields = func_get_arg(0);
-        } else {
-            $fields = func_get_args();
-        }
-        return $this->set("fetch", join(",", $fields));
+        return $this->set("fetch", implode(",", $fields));
     }
 
     /**
-     * Set the fetchLinks parameter: additional fields to retrieve for DocumentLink, You can pass in parameter
-     * an array of strings, or several strings.
+     * Set the fetchLinks parameter: additional fields to retrieve for DocumentLink
      *
-     * @return Prismic::SearchForm a clone of the SearchForm object with the new fetchLinks parameter added
+     * Pass multiple string arguments or an array of strings to unpack with the splat operator
+     *
+     * @param string[] $fields
+     * @return self
      */
-    public function fetchLinks()
+    public function fetchLinks(string ...$fields) : self
```

Comment:
could you also add graphQuery support in seach form? and of course add the test and add it in your data.json.
https://github.com/prismicio/prismic-javascript/blob/alw/graphquery/src/form.ts#L56